src/widget/wpushbutton.cpp

```diff
@@ -395,12 +425,22 @@ void WPushButton::mousePressEvent(QMouseEvent * e) {
         } else {
             // Toggle through the states
             emitValue = getControlParameterLeft();
+            const auto oldValue = emitValue;
             if (!util_isnan(emitValue) && m_iNoStates > 0) {
                 emitValue = static_cast<int>(emitValue + 1.0) % m_iNoStates;
             }
             if (m_leftButtonMode == ControlPushButton::LONGPRESSLATCHING) {
                 m_clickTimer.setSingleShot(true);
                 m_clickTimer.start(ControlPushButtonBehavior::kLongPressLatchingTimeMillis);
+                if (oldValue == 0.0) {
+                    // Capture pixmap with the off state ...
+                    m_preLongPressPixmap = QPixmap(static_cast<int>(width() * devicePixelRatio()),
+                            static_cast<int>(height() * devicePixelRatio()));
+                    m_preLongPressPixmap.setDevicePixelRatio(devicePixelRatio());
+                    paintOnDevice(&m_preLongPressPixmap);
+                    // ... and start the long press latching animation
+                    m_animTimer.start(1000 / 60);
```

Comment:
if I understand correctly, this creates an independent 60fps timer to update this animation.  Would it be possible / feasible to connect instead to the global UI update timer?  or does it not really matter because it's just updating the position data, and the drawing is done elsewhere?

Reply:
I doesn't matter. The overhead of the timer is minimal and it doesn't do the actual drawing, it just triggers the drawing when Qt sees fit. This is the recommended way.